kasa/transports/linkietransport.py

```diff
@@ -0,0 +1,137 @@
+"""Implementation of the legacy TP-Link Smart Home Protocol.
+
+Encryption/Decryption methods based on the works of
+Lubomir Stroetmann and Tobias Esser
+
+https://www.softscheck.com/en/reverse-engineering-tp-link-hs110/
+https://github.com/softScheck/tplink-smartplug/
+
+which are licensed under the Apache License, Version 2.0
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+from __future__ import annotations
+
+import logging
+import base64
+import ssl
+from urllib.parse import quote
+from yarl import URL
+
+from kasa.httpclient import HttpClient
+from kasa.deviceconfig import DeviceConfig
+from kasa.exceptions import KasaException, _RetryableError
+from kasa.json import loads as json_loads
+from kasa.credentials import DEFAULT_CREDENTIALS, Credentials, get_default_credentials
+from kasa.transports.xortransport import XorEncryption
+
+from .basetransport import BaseTransport
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LinkieTransport(BaseTransport):
+    """Implementation of the Linkie encryption protocol
+    
+    Linkie is used as the endpoint for TP-Link's camera encryption
+    protocol, used by newer firmware versions.
+    """
+
+    DEFAULT_PORT: int = 10443
+    CIPHERS = ":".join(
+        [
+            "AES256-GCM-SHA384",
+            "AES256-SHA256",
+            "AES128-GCM-SHA256",
+            "AES128-SHA256",
+            "AES256-SHA",
+        ]
+    )
+
+    def __init__(self, *, config: DeviceConfig) -> None:
+        super().__init__(config=config)
+        self._http_client = HttpClient(config)
+        self._ssl_context = self._create_ssl_context()
+        self._app_url = URL(f"https://{self._host}:{self._port}/data/LINKIE2.json")
+
+        self._headers = {
+            "Authorization": f"Basic {self.credentials_hash}",
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+
+    @property
+    def default_port(self) -> int:
+        """Default port for the transport."""
+        return self.DEFAULT_PORT
+
+    @property
+    def credentials_hash(self) -> str | None:
+        """The hashed credentials used by the transport."""
+        if self._credentials == Credentials():
+            return None
+        creds = get_default_credentials(DEFAULT_CREDENTIALS["KASACAMERA"])
+        creds_combined = f"{creds.username}:{creds.password}"
+        return base64.b64encode(creds_combined.encode()).decode()
```

Comment:
It always uses the default credentials. After linking to my TP-Link account, dumping device info requires my cloud credentials, but a direct command works without them.

Reply:
The way it should be implemented is first try connecting with the supplied credentials, then if that fails try the default credentials. That's then consistent with the other transports. I'm also not sure why `dump_devinfo` needs the cloud credentials because isn't it just using the default when it gets to the transport anyway?

Reply:
I looked into this again and discovered it simply needs  `--username` and `--password` set to *something* so the credentials hash is passed to the device. I believe all API calls using `Linkie` need the Basic Auth header.

Reply:
Needed to pass credentials because of a bug. Fixed in 3f6fc30bf3776b15ba827a57ab125bf33bf30b0d